go/consensus/tendermint/apps/roothash/roothash.go

```diff
@@ -497,6 +502,7 @@ func (app *rootHashApplication) tryFinalizeExecutor(
 		// Messages omitted on purpose.
 
 		rtState.Round.ExecutorPool.ResetCommitments()
+		rtState.Round.Transition(blk, ctx.BlockHeight())
 		rtState.Round.Finalized = true
```

Comment:
What is the point of this finalized flag now? It is set to true here and then set to false in `postProcessFinalizedBlock` (in `Reset`) which is executed immediately after? If we change the current block immediately, any further commitments will fail anyway as they will be for the wrong block/round? I think we can just get rid of the finalized flag.

Reply:
Actually why do we even copy the `CurrentBlock` into `Round` since it is already available in runtime state? We would just need to change `Round`'s `AddExecutorCommitment` to take the block as an argument?

Reply:
Didn't even notice that was the only use case for that flag, thanks! fixed

Reply:
Cool, thanks! See also my comment about `CurrentBlock` in `Round`, I think we can remove the duplicate block as well?

Reply:
> Actually why do we even copy the CurrentBlock into Round since it is already available in runtime state? 

No idea :-) 

> We would just need to change Round's AddExecutorCommitment to take the block as an argument?

Yeah can do that

Reply:
fixed as well